src/app/shipping/MultiShippingForm.tsx

```diff
@@ -87,32 +103,92 @@ class MultiShippingForm extends PureComponent<MultiShippingFormProps & WithLangu
         }
 
         return (
-            <Form>
-                <ul className="consignmentList">
-                    { items.map(item => (
-                        <li key={ item.key }>
-                            <ItemAddressSelect
-                                addresses={ addresses }
-                                item={ item }
-                                onSelectAddress={ this.handleSelectAddress }
-                                onUseNewAddress={ onUseNewAddress }
-                            />
-                        </li>
-                    )) }
-                </ul>
-
-                <ShippingFormFooter
-                    cartHasChanged={ cartHasChanged }
+            <Fragment>
+                { !!itemAddingAddress && <AddressFormModal
+                    countries={ countries }
+                    countriesWithAutocomplete={ countriesWithAutocomplete }
+                    defaultCountryCode={ defaultCountryCode }
+                    getFields={ getFields }
+                    googleMapsApiKey={ googleMapsApiKey }
                     isLoading={ isLoading }
-                    isMultiShippingMode={ true }
-                    shouldDisableSubmit={ this.shouldDisableSubmit() }
-                    shouldShowOrderComments={ shouldShowOrderComments }
-                    shouldShowShippingOptions={ !hasUnassignedLineItems(consignments, cart.lineItems) }
-                />
-            </Form>
+                    isOpen={ true }
```

Comment:
Should you just pass `itemAddingAddress` to `isOpen`? It seems to be me that it's a variable to control the visibility of the modal?

Reply:
I could do that, but I actually prefer to completely destroy the modal. The reason is that if I don't destroy the modal, the form values are remembered and the second time you open the modal to add a second address, it will remember the values from the first address.